extensions/cloudfiles-extensions/src/main/java/io/druid/storage/cloudfiles/CloudFilesStorageDruidModule.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.storage.cloudfiles;
+
+import java.util.List;
+
+import org.jclouds.ContextBuilder;
+import org.jclouds.logging.slf4j.config.SLF4JLoggingModule;
+import org.jclouds.rackspace.cloudfiles.v1.CloudFilesApi;
+
+import com.fasterxml.jackson.core.Version;
+import com.fasterxml.jackson.databind.Module;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Binder;
+import com.google.inject.Provides;
+import com.metamx.common.logger.Logger;
+
+import io.druid.guice.Binders;
+import io.druid.guice.JsonConfigProvider;
+import io.druid.guice.LazySingleton;
+import io.druid.initialization.DruidModule;
+
+/**
+ */
+public class CloudFilesStorageDruidModule implements DruidModule {
+
+	private static final Logger log = new Logger(CloudFilesStorageDruidModule.class);
+	public static final String SCHEME = "cloudfiles";
+
+	@Override
+	public List<? extends Module> getJacksonModules() {
+		log.info("Getting jackson modules...");
+
+		return ImmutableList.of(new Module() {
+			@Override
+			public String getModuleName() {
+				return "CloudFiles-" + System.identityHashCode(this);
+			}
+
+			@Override
+			public Version version() {
+				return Version.unknownVersion();
+			}
+
+			@Override
+			public void setupModule(SetupContext context) {
+				context.registerSubtypes(CloudFilesLoadSpec.class);
+			}
+		});
+	}
+
+	@Override
+	public void configure(Binder binder) {
+		log.info("Configuring CloudFilesStorageDruidModule...");
+		JsonConfigProvider.bind(binder, "druid.storage", CloudFilesDataSegmentPusherConfig.class);
+		JsonConfigProvider.bind(binder, "druid.cloudfiles", CloudFilesAccountConfig.class);
+
+		Binders.dataSegmentPullerBinder(binder).addBinding(SCHEME).to(CloudFilesDataSegmentPuller.class)
+				.in(LazySingleton.class);
+		Binders.dataSegmentPusherBinder(binder).addBinding(SCHEME).to(CloudFilesDataSegmentPusher.class)
+				.in(LazySingleton.class);
+
+		log.info("Configured CloudFilesStorageDruidModule.");
+	}
+
+	@Provides
+	@LazySingleton
+	public CloudFilesApi getCloudFilesApi(final CloudFilesAccountConfig config) {
+		Iterable<com.google.inject.Module> modules = ImmutableSet.<com.google.inject
```

Comment:
this should probably be either included at the pom level or set as a command line parameter for jcloud.